### PR TITLE
Implemented OMS importer package

### PR DESCRIPTION
**Pull request by HighQualityWaschbaer**

### CODE REVIEW

File: Application/pkg/config/config.go [hunk 1]
```diff
@@ -4,21 +4,18 @@ import (
...
 
 var instance *Config
 
 type Config struct {
-	Port      string `yaml:"port"`
-	PathToDB  string `yaml:"pathToDB"`
-	PathForDB string `yaml:"pathForDB"`
+	Port       string `yaml:"port"`
+	OMSDBPath  string `yaml:"OMSDBPath"`
+	GOSTDBPath string `yaml:"GOSTDBPath"`
```
Comment (Valentin-Kaiser) [inline Application/pkg/config/config.go:15]: Inconsistent naming of variables. Check upper/lower case.

File: Application/pkg/importer/imports/converter/converter.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func IdToUint(value interface{}) interface{} {
+	id, err := strconv.Atoi(fmt.Sprint(value))
+	if err != nil {
+		return 999
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/converter/converter.go:11]: Why is 999 returned as value in case of error?

Comment (HighQualityWaschbaer) [reply Application/pkg/importer/imports/converter/converter.go:11]: 999 is the designated id for "Sonstige"(other). This was specifically requested in the original importer software.

File: Application/pkg/config/config.go (hunk 1 above)
Comment (Plaenkler) [reply Application/pkg/config/config.go]: Updated in [31a8d6c](https://github.com/Plaenkler/GoST/pull/5/commits/31a8d6ca6ba3250af116883705a6ec743fb243bc)

File: Application/pkg/importer/imports/workshopInvoices.go [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+	}
+	defer wiDbf.Close()
+
+	invPosDbf, err := dbase.Open(filepath.Join(OMSDBPath, "KUNDRECP.dbf"), &dbase.Win1250Converter{})
+	if err != nil {
+		return err
+	}
+	defer invPosDbf.Close()
+
+	//import workshop invoices
```
Comment (Plaenkler) [inline Application/pkg/importer/imports/workshopInvoices.go:40]: The comments should look like // *. //* is not allowed. Furthermore, capitalization is completely omitted.

Comment (Plaenkler) [inline Application/pkg/importer/imports/workshopInvoices.go:40]: The comments should look like // *. //* is not allowed. Furthermore, capitalization is completely omitted.

**Review by Plaenkler — CHANGES_REQUESTED** [review]

File: Application/pkg/importer/imports/workshopInvoices.go (hunk 3 above)
Comment (Plaenkler) [reply Application/pkg/importer/imports/workshopInvoices.go]: Solved in [1bbcba3](https://github.com/Plaenkler/GoST/pull/5/commits/1bbcba3da2034955576d452e1e252c18f4a987b8)

**Review by Plaenkler — APPROVED** [review]

**Review by HighQualityWaschbaer — COMMENTED** [review]
I can't approve this pull request as requested because it is mine but i approve this pr.

File: Application/pkg/importer/imports/agents.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	dbf.SetTrimspacesDefault(true)
+	dbf.SetColumnModification(dbf.ColumnPos("KZ"), true, "ID", nil)
+	dbf.SetColumnModification(dbf.ColumnPos("NAME"), true, "Name", nil)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			return err
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/agents.go:30]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/articles.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	dbf.SetColumnModification(dbf.ColumnPos("EKPREIS"), true, "PurchasePrice", nil)
+	dbf.SetColumnModification(dbf.ColumnPos("BENENNUNG"), true, "Name", nil)
+	dbf.SetColumnModification(dbf.ColumnPos("WG2"), true, "ManufacturerID", converter.IdToUint)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			return err
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/articles.go:32]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/customerErrors.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	dbf.SetTrimspacesDefault(true)
+	dbf.SetColumnModification(dbf.ColumnPos("KZ"), true, "ID", nil)
+	dbf.SetColumnModification(dbf.ColumnPos("BENENNUNG"), true, "Description", nil)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			return err
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/customerErrors.go:29]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/detectedErrors.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	dbf.SetTrimspacesDefault(true)
+	dbf.SetColumnModification(dbf.ColumnPos("KZ"), true, "ID", nil)
+	dbf.SetColumnModification(dbf.ColumnPos("BENENNUNG"), true, "Description", nil)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			return err
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/detectedErrors.go:29]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/manufacturers.go [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	dbf.SetTrimspacesDefault(true)
+	dbf.SetColumnModification(dbf.ColumnPos("KZ"), true, "ID", converter.IdToUint)
+	dbf.SetColumnModification(dbf.ColumnPos("TEXT"), true, "Name", nil)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			return err
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/manufacturers.go:30]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/technicians.go [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	dbf.SetTrimspacesDefault(true)
+	dbf.SetColumnModification(dbf.ColumnPos("TECHNR"), true, "ID", nil)
+	dbf.SetColumnModification(dbf.ColumnPos("NAME"), true, "Name", nil)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			return err
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/technicians.go:29]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/workshopInvoices.go [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+
+	// import and map regular invoice data
+	invoicesData := importInvoiceData(OMSDBPath)
+
+	for !wiDbf.EOF() {
+		row, err := wiDbf.Row()
+		if err != nil {
+			return err
+		}
+		wiDbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/workshopInvoices.go:55]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/workshopInvoices.go [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+	if err != nil {
+		log.Panicf(err.Error())
+	}
+	defer dbf.Close()
+
+	invData := make(map[string]invoiceData, 0)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/workshopInvoices.go:166]: Error return value of dbf.Skip is not checked

File: Application/pkg/importer/imports/workshopInvoices.go [hunk 12]
```diff
@@ -0,0 +1,265 @@
...
+
+	db := database.GetManager().DB
+	id := uint64(1)
+
+	for !dbf.EOF() {
+		row, err := dbf.Row()
+		if err != nil {
+			log.Panicf(err.Error())
+		}
+		dbf.Skip(1)
```
Comment (Valentin-Kaiser) [inline Application/pkg/importer/imports/workshopInvoices.go:242]: Error return value of dbf.Skip is not checked

**Pull request merged**